### PR TITLE
Put service key binding/instance creation and job creation in one transaction

### PR DESCRIPTION
**Pull request by kathap**
Similar as in #3567, now for service key binding and service instance creation to be consistent with #3567.

Thanks for contributing to cloud_controller_ng. To speed up the process of reviewing your pull request please provide us with:

* Links to any other associated PRs
#3567
* [X] I have reviewed the [contributing guide](https://github.com/cloudfoundry/cloud_controller_ng/blob/main/CONTRIBUTING.md)

* [X] I have viewed, signed, and submitted the Contributor License Agreement

* [X] I have made this pull request to the `main` branch

* [X] I have run all the unit tests using `bundle exec rake`

* [ ] I have run [CF Acceptance Tests](https://github.com/cloudfoundry/cloud_controller_ng/blob/main/spec/README.md#cf-acceptance-tests-cats)

### CODE REVIEW

File: app/controllers/v3/service_credential_bindings_controller.rb [hunk 1]
```diff
@@ -182,10 +181,12 @@ def override_default_order_by(message)
...
     action = V3::ServiceCredentialBindingKeyCreate.new(user_audit_info, message.audit_hash)
-    binding = action.precursor(service_instance, message:)
+    VCAP::CloudController::ServiceBinding.db.transaction do
+      binding = action.precursor(service_instance, message:)
 
-    pollable_job_guid = enqueue_bind_job(:key, binding.guid, message)
-    head :accepted, 'Location' => url_builder.build_url(path: "/v3/jobs/#{pollable_job_guid}")
+      pollable_job_guid = enqueue_bind_job(:key, binding.guid, message)
+      head :accepted, 'Location' => url_builder.build_url(path: "/v3/jobs/#{pollable_job_guid}")
+    end
```
[the remark this answers is not in the document]
Comment (kathap) [reply app/controllers/v3/service_credential_bindings_controller.rb:189]: this = when db is unavailable to raise the appropriate error?

File: spec/request/service_instances_spec.rb [hunk 2]
```diff
@@ -1174,6 +1174,23 @@ def check_filtered_instances(*instances)
         end
       end
 
+      describe 'when db is unavailable' do
+        before do
+          allow_any_instance_of(VCAP::CloudController::Jobs::Enqueuer).to receive(:enqueue_pollable).and_raise(Sequel::DatabaseDisconnectError)
+        end
+
+        it 'rolls back the transaction' do
```
[the remark this answers is not in the document]
Comment (kathap) [reply spec/request/service_instances_spec.rb:1182]: Ah, true!

File: spec/request/service_credential_bindings_spec.rb [hunk 3]
```diff
@@ -1601,6 +1601,12 @@ def check_filtered_bindings(*bindings)
                                                                    'code' => 10_015
                                                                  }))
           end
+
+          it 'rolls back the transaction' do
+            api_call.call(admin_headers)
+
+            expect(service_instance.service_keys.count).to eq(0)
```
Comment (philippthun) [inline spec/request/service_credential_bindings_spec.rb:1608]: Here you have to check `service_instance.service_bindings`. Service keys are always 0 even without the transaction.

File: spec/request/service_route_bindings_spec.rb [hunk 4]
```diff
@@ -1006,6 +1006,12 @@
                                                                  'code' => 10_015
                                                                }))
         end
+
+        it 'rolls back the transaction' do
+          api_call.call(admin_headers)
+
+          expect(service_instance.service_keys.count).to eq(0)
```
Comment (philippthun) [inline spec/request/service_route_bindings_spec.rb:1013]: Here you have to check `service_instance.routes`. Service keys are always 0 even without the transaction.

Comment (kathap) [reply spec/request/service_route_bindings_spec.rb:1013]: Oh, good point

**Pull request merged**